jsapp/js/account/plan.component.tsx

```diff
@@ -0,0 +1,339 @@
+      .always(() =>
+        setButtonDisabled(!buttonsDisabled)
+      );
+  };
+
+  const managePlan = () => {
+    console.log('Manage', state.organization?.uid);
+    if (!state.organization?.uid || buttonsDisabled){
+      return;
+    }
+    setButtonDisabled(buttonsDisabled);
+    $.ajax({
+      dataType: 'json',
+      method: 'POST',
+      url: `${ROOT_URL}/api/v2/stripe/customer-portal?organization_uid=${state.organization?.uid}`,
+    })
+      .done(function (res) {
+        window.location.assign(res.url);
+      })
+      .fail(handleApiFail)
+      .always(() =>
+        setButtonDisabled(!buttonsDisabled)
+      );
+  };
+
+  // Get feature items and matching icon boolean
+  const getListItem = (listType: string, plan: string) => {
+    const listItems: {icon: boolean, item: any}[] = [];
+    filterPrices().map((price: any) => (
+      Object.keys(price.metadata).map((featureItem: any) => {
+        const numberItem = featureItem.lastIndexOf('_');
+        const currentResult = featureItem.substring(numberItem + 1);
+        switch (listType) {
+          case 'support':
+            const currentIcon = `feature_support_check_${currentResult}`;
+            if (featureItem.includes('feature_support_') && !featureItem.includes('feature_support_check') && price.name === plan) {
+              const keyName = `feature_support_${currentResult}`;
+              let iconBool = false;
+              const itemName: string = price.metadata[keyName];
+              if (price.metadata[currentIcon] !== undefined) {
+                iconBool = JSON.parse(price.metadata[currentIcon]);
+                listItems.push({icon: iconBool, item: itemName});
+              }
+            }
+            break;
+          case 'advanced':
+            const currIcon = `feature_advanced_check_${currentResult}`;
+            if (featureItem.includes('feature_advanced_') && !featureItem.includes('feature_advanced_check') && price.name === plan) {
+              const keyName = `feature_advanced_${currentResult}`;
+              let iconBool = false;
+              const itemName: string = price.metadata[keyName];
+              if (price.metadata[currIcon]) {
+                iconBool = JSON.parse(price.metadata[currIcon]);
+              }
+              listItems.push({icon: iconBool, item: itemName});
+            }
+            break;
+          case 'addons':
+            const currentItemIcon = `feature_addons_check_${currentResult}`;
+            if (featureItem.includes('feature_addons_') && !featureItem.includes('feature_addons_check') && price.name === plan) {
+              const keyName = `feature_addons_${currentResult}`;
+              let iconBool = false;
+              const itemName: string = price.metadata[keyName];
+              if (price.metadata[currentItemIcon]) {
+                iconBool = JSON.parse(price.metadata[currentItemIcon]);
+              }
+              listItems.push({icon: iconBool, item: itemName});
```

Comment:
There's a lot of repeated logic and short lived variables like iconBool. I think this could be refactored. But I'd have to play around with it to give more specific advice.